Packages/com.unity.multiplayer.samples.coop/Tests/Runtime/HostAndDisconnectTest.cs

```diff
@@ -0,0 +1,265 @@
+
+            var scope = clientMainMenuState.DIScope;
+
+            var lobbyUIMediator = scope.Resolve<LobbyUIMediator>();
+            Assert.That(lobbyUIMediator != null, $"{nameof(LobbyUIMediator)} component not found!");
+
+            var lobbyCreationUI = lobbyUIMediator.LobbyCreationUI;
+            Assert.That(lobbyCreationUI != null, $"{nameof(LobbyCreationUI)} component not found!");
+
+            // validate unity services login successful
+
+            // wait until authenticated
+            var timer = k_ServiceQueryTimeout;
+            while (timer > 0f && !AuthenticationService.Instance.IsAuthorized)
+            {
+                timer -= Time.deltaTime;
+                yield return null;
+            }
+
+            clientMainMenuState.OnStartClicked();
+
+            yield return new WaitForEndOfFrame();
+
+            lobbyUIMediator.ToggleCreateLobbyUI();
+
+            // a confirmation popup will appear; wait a frame for it to pop up
+            yield return new WaitForEndOfFrame();
+
+            lobbyCreationUI.OnCreateClick();
+
+            // get LobbyServiceFacade
+            var applicationController = GameObject.FindObjectOfType<ApplicationController>();
+            Assert.That(applicationController != null, $"{nameof(ApplicationController)} component not found!");
+
+            var lobbyServiceFacade = applicationController.LobbyServiceFacade;
+            Assert.That(lobbyServiceFacade != null, $"{nameof(LobbyServiceFacade)} component not found!");
+
+            // wait until lobby has been created
+            timer = k_ServiceQueryTimeout;
+            while (timer > 0f && lobbyServiceFacade.CurrentUnityLobby == null)
+            {
+                timer -= Time.deltaTime;
+                yield return null;
+            }
+
+            Assert.NotNull(lobbyServiceFacade.CurrentUnityLobby);
+
+            // lobby has been created; now relay data will be allocated; wait until lobby relay data updated
+            timer = k_ServiceQueryTimeout;
+            while (timer > 0f && string.IsNullOrEmpty(lobbyUIMediator.LocalLobby.RelayJoinCode))
+            {
+                timer -= Time.deltaTime;
+                yield return null;
+            }
+
+            Assert.IsTrue(!string.IsNullOrEmpty(lobbyUIMediator.LocalLobby.RelayJoinCode));
+
+            // lobby creation will initialize the hosting process; next frame the results will be ready
+            yield return null;
+
+            // verify hosting is successful
+            Assert.That(m_NetworkManager.IsListening);
+
+            // CharSelect is loaded as soon as hosting is successful, validate it is loaded
+            yield return GoToCharacterSelection(playerIndex);
+
+            yield return GoToBossRoomScene();
+
+            yield return Disconnect();
+        }
+
+        IEnumerator GoToMainMenuScene()
+        {
+            // load Bootstrap scene
+            SceneManager.LoadSceneAsync(k_BootstrapSceneName);
```

Comment:
I'm a bit confused at what we're trying to test here.
The actual in game flow would use LP's scene wrapper, why the direct call here?
If we're doing unit tests, we should call lower level methods individually, with no dependencies and make sure to only test that unit.
If we're doing integration tests, we should use as little custom code to move things around and rely on in-game flow code already present in game.